src/legacy/ui/public/chrome/directives/kbn_chrome.js

```diff
@@ -77,9 +77,9 @@ export function kbnChromeProvider(chrome, internals) {
           $rootScope.$on('$locationChangeStart', (e, newUrl) => {
             // This handler fixes issue #31238 where browser back navigation
             // fails due to angular 1.6 parsing url encoded params wrong.
-            const absUrlHash = url.parse($location.absUrl()).hash;
```

Comment:
I still think it might be a good idea to scope this to explicit plugins. Angular routing is already causing a lot of headache downstream as-is, so adding this will make it even harder to debug when something is going awry.

Reply:
I'm not totally opposed to that, but this is a problem that can happen anywhere so what plugins should we limit this to? I agree that broad changes like this are undesirable, but I'd rather have it be opt-out for plugins that find it unnecessary or problematic in some way.

To accomplish this I've added the `chrome.disableAutoAngularUrlEncodingFix()` method to `ui/chrome`, which when called at initialization will prevent the event handler from being registered. I've also [implemented this in APM](https://github.com/elastic/kibana/blob/fba3d3e4361be98ce8a5f0eb45da29bf8dd4055e/x-pack/plugins/apm/public/index.tsx#L41).

Note that this means going to a URL like `http://localhost:5601/wzz/app/apm#/services?_g=()&b=c/d` breaks the history in APM as the back button will take you to `...&b=c/d` and then immediately back to `...&b=c%2Fd` thanks to the fact that angular services like `GlobalState` are updating the url with the angular services, which use angular encoding...

Reply:
> thanks to the fact that angular services like GlobalState are updating the url with the $location service, which uses angular encoding

Okay, didn't think of that but makes sense. We might end up not using `disableAutoAngularUrlEncodingFix` if it doesn't make sense to disable as you say.

Slightly off topic: I previously looked into disabling GlobalState for APM. Would that be a bad idea?

Reply:
If you're not using it then I don't see why it would be a problem.

Reply:
> If you're not using it then I don't see why it would be a problem.

Isn't this fix a pretty good example of a problem where the root cause is GlobalState (or did I misunderstand that)?
I'd rather not have those problems in the first place. As you say, `GlobalState` is rewriting the url which can cause problems with the history stack (and thus the back button). Eg. Currently I'm seeing `_g` being added to the url for no good reason.

Reply:
Yeah, the behavior of global state is similar to a lot of our angular behavior; simply by loading the file in the browser new behaviors are introduced which impact a lot of unsuspecting code. This is a problem but one we're probably not going to fix by making changes to the state services, but by reachitecting the way we do state management in the URL in the new platform... At least that's my guess, but I'm open to idea or suggestions about how to make this better.

Reply:
> This is a problem but one we're probably not going to fix by making changes to the state services, but by reachitecting the way we do state management in the URL in the new platform... At least that's my guess, but I'm open to idea or suggestions about how to make this better.

Super glad to hear this is on your radar for the new platform. I'm hoping that any kind of global state will be opt-in, so a plugin have full autonomy over the url path that comes after the base path (what is today `/app/<plugin>`).